boards/nrf51dk/Makefile

```diff
@@ -30,5 +32,39 @@ clean:
 .PHONY: debug
 debug: target/nrf51/debug/nrf51dk
 
-#TODO: Figure out how to program the nRF51dk
+.PHONY: apps/$(APP)/build/$(TOCK_ARCH)/app.bin
+apps/$(APP)/build/$(TOCK_ARCH)/app.bin:
+	@make -C apps/$(APP) TOCK_ARCH=$(TOCK_ARCH)
+
+target/nrf51/release/nrf51dk-$(APP): target/nrf51/release/nrf51dk apps/$(APP)/build/$(TOCK_ARCH)/app.bin
```

Comment:
I'm also a little concerned that making an apps folder inside `boards/nrf51dk/` will be confusing to users.